src/applications/facility-locator/containers/FacilitesMap.jsx

```diff
@@ -0,0 +1,469 @@
+            facilityType: props.currentQuery.facilityType,
+            serviceType: props.currentQuery.serviceType,
+            page: resultsPage,
+          });
+          setIsSearching(false);
+        }
+        if (searchResultTitleRef.current) {
+          setFocus(searchResultTitleRef.current);
+        }
+      }
+      return () => {
+        // remove the listeners
+      };
+    },
+    [props.currentQuery.id], // Handle search when query changes
+  );
+
+  const renderMarkers = locations => {
+    if (locations.length === 0) return;
+    const currentLocation = props.currentQuery.position;
+    const markersLetters = MARKER_LETTERS.values();
+    const sortedLocations = locations
+      .map(r => {
+        const distance = currentLocation
+          ? distBetween(
+              currentLocation.latitude,
+              currentLocation.longitude,
+              r.attributes.lat,
+              r.attributes.long,
+            )
+          : null;
+        return {
+          ...r,
+          distance,
+        };
+      })
+      .sort((resultA, resultB) => resultA.distance - resultB.distance);
+
+    sortedLocations.forEach(loc => {
+      const attrs = {
+        letter: markersLetters.next().value,
+      };
+      const markerElement = buildMarker('location', { loc, attrs });
+      new mapboxgl.Marker(markerElement)
+        .setLngLat([loc.attributes.long, loc.attributes.lat])
+        .addTo(map);
+    });
+
+    if (props.currentQuery.bounds) {
+      map.fitBounds(props.currentQuery.bounds); // {duration: 0} to disable animation
+    }
+    if (props.currentQuery.searchCoords) {
+      const markerElement = buildMarker('currentPos');
+      new mapboxgl.Marker(markerElement)
+        .setLngLat([
+          props.currentQuery.searchCoords.lng,
+          props.currentQuery.searchCoords.lat,
+        ])
+        .addTo(map);
+    }
+  };
+
+  useEffect(
+    () => {
+      if (!map) return;
+      clearLocationMarkers();
+      renderMarkers(props.results);
+    },
+    [props.results], // Handle build markers when we get results
+  );
+
+  const handleSearch = async () => {
+    const { currentQuery } = props;
+    currentZoom = null;
+
+    updateUrlParams({
+      address: currentQuery.searchString,
+    });
+
+    props.genBBoxFromAddress({
+      ...currentQuery,
+    });
+
+    setIsSearching(true);
+  };
+
+  const handlePageSelect = page => {
+    const { currentQuery } = props;
+    props.searchWithBounds({
+      bounds: currentQuery.bounds,
+      facilityType: currentQuery.facilityType,
+      serviceType: currentQuery.serviceType,
+      page,
+    });
+  };
+
+  const setupMap = (setMapInit, mapContainerInit) => {
+    mapboxgl.accessToken = mapboxToken;
+    const mapInit = new mapboxgl.Map({
+      container: mapContainerInit,
+      style: 'mapbox://styles/mapbox/outdoors-v11',
+      center: [-99.27246093750001, 40.17887331434698], // Initial state search query reducer
+      zoom: 3,
```

Comment:
If we need to specify a hard-coded Zoom level, can we try making it very high, so fitBounds adjusts it as needed?

Reply:
these values is o that it centers the map on the us at initial render, I will put them in constants.

Reply:
Thanks. As I mentioned elsewhere, those same coordinates are used in several places, so yes, please put them in a const.